OpenRA.Mods.Common/Graphics/ContrailRenderable.cs

```diff
@@ -30,8 +30,14 @@ public class ContrailRenderable : IRenderable, IFinalizedRenderable
 		int length;
 		int skip;
 
-		public ContrailRenderable(World world, Color color, WDist width, int length, int skip, int zOffset)
-			: this(world, new WPos[length], width, 0, 0, skip, color, zOffset) { }
+		public ContrailRenderable(World world, Color color, WDist width, int length, int skip, int zOffset, WPos source)
+			: this(world, new WPos[length], width, 0, 0, skip, color, zOffset)
+		{
+			// Make sure we have enough segments to draw on the first frame.
+			// Otherwise fast actors will have their trail not starting at their origin.
+			for (var i = 0; i < 3; i++)
```

Comment:
I wonder if that wants to be `< 4` though, since otherwise the first `Render` call will not draw anything(?). (This only adds three entries right now, correct? Then after one tick/update we will have four entries. That will still make the loop in `Render` not run because `i < length - skip - 4` is always false for `i = 0`, `length <= 4` and `skip = 0`? The `// Need at least 4 points to smooth the contrail over` comment might actually be wrong and we need 5 points?)

Reply:
The idea was that it needs to have ticked once, otherwise there wont be any position update which generated a contrail.

Reply:
Point is that it needs to tick twice to draw anything iirc.